### PR TITLE
fix(VField/VInput): centerAffix doesn't work for underlined/plain

### PR DESCRIPTION
**Pull request by yuwu9145**
Automatically apply singleLine under the hood, and vertically align **appendInner**, **prependInner**, **clearIcon**, **label** and **input field** in the center.

## Markup:

```vue
<template>
  <v-container class="pa-md-12 bg-grey">
    <v-row>
      <v-col cols="4">
        <v-text-field
          bg-color="white"
          prepend-icon="$vuetify"
          prepend-inner-icon="$vuetify"
          placeholder="solo"
          label="solo"
          variant="solo"
          center-affix
        />
        <v-text-field
          bg-color="white"
          prepend-icon="$vuetify"
          prepend-inner-icon="$vuetify"
          placeholder="outlined"
          label="outlined"
          variant="outlined"
          center-affix
        />
        <v-text-field
          bg-color="white"
          prepend-icon="$vuetify"
          prepend-inner-icon="$vuetify"
          placeholder="filled"
          label="filled"
          variant="filled"
          center-affix
        />
        <v-text-field
          bg-color="white"
          clearable
          prepend-icon="$vuetify"
          prepend-inner-icon="$vuetify"
          placeholder="underlined"
          label="underlined"
          variant="underlined"
          center-affix
        />
      </v-col>
      <v-col cols="4">
        <v-text-field
          bg-color="white"
          prepend-icon="$vuetify"
          prepend-inner-icon="$vuetify"
          placeholder="solo"
          label="solo"
          variant="solo"
        />
        <v-text-field
          bg-color="white"
          prepend-icon="$vuetify"
          prepend-inner-icon="$vuetify"
          placeholder="outlined"
          label="outlined"
          variant="outlined"
        />
        <v-text-field
          bg-color="white"
          prepend-icon="$vuetify"
          prepend-inner-icon="$vuetify"
          placeholder="filled"
          label="filled"
          variant="filled"
        />
        <v-text-field
          bg-color="white"
          clearable
          prepend-icon="$vuetify"
          prepend-inner-icon="$vuetify"
          placeholder="underlined"
          label="underlined"
          variant="underlined"
        />
      </v-col>
    </v-row>
  </v-container>
</template>

<script setup>
  import { ref } from 'vue'
</script>




```

### CODE REVIEW

**Comment by johnleider** [comment]
Is there a collection of props that would move the label to the middle as well?
image

**Comment by yuwu9145** [comment]
> Is there a collection of props that would move the label to the middle as well? 

Need some additional fix to make sure label/input centre-aligned for underlined variant, same as v2 does
[v2 demo](https://codepen.io/yuchaosydney/pen/mdYgNJr?editors=101)
[v3 demo](https://play.vuetifyjs.com/#eNpVj88OgjAMh19l2VnkbiaJd99APBQoZEkpSxlEY3x3tylxXPbn+23t19tLz9KWF+eO64L6pI3H0RF4rGpWyqwFOJeO6dJO7MEyyg8l6PHhi94idRtUiqBBOtf6Gvda/4MVxAL7EC3coVAo1uV5M4QmNEl44MSOIM88nS0PhEX8tcHKlLnC5hrgXjaS7yymzGaMFn2SYP0+8EKUlvsHLDtZbw==)

**Comment by yuwu9145** [comment]
@johnleider See if you were satisfied with the API description/usage change, I will handle the technical part and make sure no regression

**Comment by johnleider** [comment]
Looks good. Not related to anything you've done, but the prepend-inner icon is really tight on the edge of the input. Is that by design or not intentional?

**Comment by yuwu9145** [comment]
I think underlined variant is supposed to extend the content to edges, we have a little gap before prepend-inner, which is ok I guess.

Use slot to add a manual gap

Screenshot 2024-07-03 at 3 00 43 pm


```vue
<v-text-field
          bg-color="white"
          clearable
          prepend-icon="$vuetify"
          placeholder="underlined"
          label="underlined"
          variant="underlined"
          center-affix
        >
          <template #prepend-inner>
            <v-icon class="ml-3" icon="$vuetify" />
          </template>
        </v-text-field>
```

File: packages/vuetify/src/components/VInput/VInput.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ export const makeVInputProps = propsFactory({
   appendIcon: IconValue,
   centerAffix: {
     type: Boolean,
-    default: true,
+    default: false,
```
Comment (johnleider) [inline packages/vuetify/src/components/VInput/VInput.tsx:45]: If default is false, can just do `centerAffix: Boolean` which defaults to false.

File: packages/vuetify/src/components/VField/VField.tsx [hunk 2]
```diff
@@ -64,7 +64,7 @@ export const makeVFieldProps = propsFactory({
   active: Boolean,
   centerAffix: {
     type: Boolean,
-    default: undefined,
+    default: false,
```
Comment (johnleider) [inline packages/vuetify/src/components/VField/VField.tsx:67]: If default is false, can just do `centerAffix: Boolean` which defaults to false.

**Review by johnleider — CHANGES_REQUESTED** [review]

**Review by johnleider — APPROVED** [review]

**Pull request merged**